test/unit/modules/transport.js

```diff
@@ -1028,550 +1220,1161 @@ describe('transport', () => {
+					expect(result)
+						.to.have.property('success')
+						.which.equals(true);
+					expect(result)
+						.to.have.property('height')
+						.which.equals(headers.height);
+					expect(result)
+						.to.have.property('broadhash')
+						.which.equals(headers.broadhash);
+					expect(result)
+						.to.have.property('httpPort')
+						.which.equals(headers.httpPort);
+					expect(result)
+						.to.have.property('version')
+						.which.equals(headers.version);
+					return expect(result)
+						.to.have.property('os')
+						.which.equals(headers.os);
+				});
+			});
 
-					it('should call __private.receiveSignatures with query');
+			describe('postSignature', () => {
+				beforeEach(done => {
+					query = {
+						signature: SAMPLE_SIGNATURE_1,
+					};
+					__private.receiveSignature = sinonSandbox.stub().callsArg(1);
+					transportInstance.shared.postSignature(query, (err, res) => {
+						error = err;
+						result = res;
+						done();
+					});
+				});
 
-					describe('when __private.receiveSignatures fails', () => {
-						it('should call callback with error = null');
+				it('should call __private.receiveSignature with query.signature as argument', () => {
+					return expect(__private.receiveSignature.calledWith(query.signature))
+						.to.be.true;
+				});
 
-						it(
-							'should call callback with result = {success: false, message: err}'
-						);
+				describe('when __private.receiveSignature succeeds', () => {
+					it('should invoke callback with object { success: true }', () => {
+						expect(error).to.equal(null);
+						return expect(result)
+							.to.have.property('success')
+							.which.is.equal(true);
 					});
+				});
 
-					describe('when __private.receiveSignatures succeeds', () => {
-						it('should call callback with error = null');
+				describe('when __private.receiveSignature fails', () => {
+					var receiveSignatureError = 'Invalid signature body ...';
 
-						it('should call callback with result = {success: true}');
+					beforeEach(done => {
+						query = {
+							signature: SAMPLE_SIGNATURE_1,
+						};
+						__private.receiveSignature = sinonSandbox
+							.stub()
+							.callsArgWith(1, receiveSignatureError);
+						transportInstance.shared.postSignature(query, (err, res) => {
+							error = err;
+							result = res;
+							done();
+						});
+					});
+
+					it('should invoke callback with object { success: false, message: err }', () => {
+						expect(error).to.equal(null);
+						expect(result)
+							.to.have.property('success')
+							.which.is.equal(false);
+						return expect(result)
+							.to.have.property('message')
+							.which.is.equal(receiveSignatureError);
 					});
 				});
+			});
 
-				describe('when query.signature is undefined', () => {
-					it('should call __private.receiveSignature');
+			describe('postSignatures', () => {
+				describe('when library.schema.validate succeeds', () => {
+					beforeEach(done => {
```

Comment:
These 2 lines might be extracted and close in common `beforeEach` section of `postSignatures`
```
				query = {
							signatures: [SAMPLE_SIGNATURE_1],
						};
						__private.receiveSignatures = sinonSandbox.stub();
```